src/main/java/io/nats/client/support/JwtUtils.java

```diff
@@ -0,0 +1,145 @@
+                long unixSeconds = exp.toMillis() / 1000;
+                JsonUtils.addField(sb, "exp", unixSeconds);
+            }
+            JsonUtils.addField(sb, "iat", iat);
+            JsonUtils.addFieldEvenEmpty(sb, "jti", jti);
+            JsonUtils.addField(sb, "iss", iss);
+            JsonUtils.addField(sb, "name", name);
+            JsonUtils.addField(sb, "nats", nats);
+            JsonUtils.addField(sb, "sub", sub);
+
+            return endJson(sb).toString();
+        }
+    }
+
+    static final String ENCODED_CLAIM_HEADER =
+            ToBase64Url("{\"typ\":\"JWT\", \"alg\":\"ed25519-nkey\"}");
+
+    static final String NATS_USER_JWT_FORMAT = "-----BEGIN NATS USER JWT-----\n" +
+            "%s\n" +
+            "------END NATS USER JWT------\n" +
+            "\n" +
+            "************************* IMPORTANT *************************\n" +
+            "    NKEY Seed printed below can be used to sign and prove identity.\n" +
+            "    NKEYs are sensitive and should be treated as secrets.\n" +
+            "\n" +
+            "-----BEGIN USER NKEY SEED-----\n" +
+            "%s\n" +
+            "------END USER NKEY SEED------\n" +
+            "\n" +
+            "*************************************************************";
+
+    public static String issueUserCreds(String accountSeed, String issuerAccount, String userSeed, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String jwt = issueUserJWT(accountSeed, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+        return String.format(NATS_USER_JWT_FORMAT, jwt, userSeed);
+    }
+
+    public static String issueUserCreds(NKey accountSigningKey, String issuerAccount, String userSeed, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String jwt = issueUserJWT(accountSigningKey, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+        return String.format(NATS_USER_JWT_FORMAT, jwt, userSeed);
+    }
+
+    public static String issueUserJWT(String accountSeed, String issuerAccount, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        NKey accountSigningKey = NKey.fromSeed(accountSeed.toCharArray());
+        return issueUserJWT(accountSigningKey, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+    }
+
+    public static String issueUserJWT(NKey accountSigningKey, String issuerAccount, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String accSigningKeyPub = new String(accountSigningKey.getPublicKey());
+
+        Claim claim = new Claim();
```

Comment:
Should these raise an exception of sorts? 
* signingKey, is a mandatory account nkey pair to sign the generated jwt.
 * accountId, is a mandatory public account nkey. Will return error when not set or not account nkey.
 * publicUserKey, is a mandatory public user nkey. Will return error when not set or not user nkey.

Reply:
Yes. All of these are now validated and have unit tests for the validation.